watertap/core/membrane_channel_base.py

```diff
@@ -639,7 +639,7 @@ def _add_calculated_pressure_change_mass_transfer_components(self):
         # `_add_calculated_pressure_change` *and/or*
         # `_add_calculated_mass_transfer_coefficient`.
         # Therefore, we add this simple guard against it being called twice.
-        if hasattr(self, "channel_height"):
+        if hasattr(self, "dh"):
```

Comment:
I think it's because if hasattr(self, "channel_height"): return was preventing the _add_calculated_pressure_change_mass_heat_transfer_components from being called twice. But now that the channel height is added to the geometry method of the channel control volume, it would be created and incorrectly prevent the _add_calculated_pressure_change_mass_heat_transfer_components from being called. So the safeguard variable has changed to dh.

Reply:
Ah ok, thank you for sorting the logic there!

Reply:
Exactly @ElmiraShamlou ! I did have some internal debate with myself about where to add the volume constraint, but this is what I arrived at. I assume we will find out later if this was a good or bad idea as we progress with efforts on implementing the dynamic RO unit.